### PR TITLE
Airbnb

### PR DESCRIPTION
**Pull request by matt-sm**
- switched the full set of rules on with basically 2 exceptions:
  - jsx file extensions
  - objects as proptypes
- couple of files copied from src folder were too messed up and I found it safer to just disable rules
- some shared form components are not being used in our code and do not have tests so impossible to know if changes work
- there are still some inconsistencies in the app folder that I'll fix in a future branch

### CODE REVIEW

File: app/components/shared/form/Datefield.js [hunk 1]
```diff
@@ -29,6 +24,12 @@ class Datefield extends React.Component {
     }
   }
 
+  state = {
```
Comment (Blackfry) [inline app/components/shared/form/Datefield.js]: https://facebook.github.io/react/docs/react-component.html#constructor.
> The constructor for a React component is called before it is mounted.
If you want to have state ready for when the component is mounted, you would set them within the `constructor` method.

Comment (matt-sm) [reply app/components/shared/form/Datefield.js]: this is the rule: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/sort-comp.md

I don't see any example in the Facebook docs that have a separate state() function as a class member so I feel like it should be deleted

Comment (matt-sm) [reply app/components/shared/form/Datefield.js]: This is an example of a component changing with no tests that is not used in app folder

File: app/components/validators.js [hunk 2]
```diff
@@ -15,9 +17,7 @@ export const required = val => {
   }
 
   if (val !== null && typeof val === 'object') {
-    return values(val).filter(function(v) {
-      return v
-    }).length
+    return values(val).filter(v => v).length
```
Comment (Blackfry) [inline app/components/validators.js]: not related to your change but I think you could remove `.filter(v => v)` as it's returning 1 for 1

**Pull request merged**